### ISSUE TITLE
New RSpec integration doesn't work unless `rails_helper` is required or Rails environment is loaded first

### ISSUE THREAD

**Issue by iMacTia** [rubyatscale/stimpack#33]
Adding `--require stimpack/rspec` to my `.rspec` raises the following error:

```
Failure/Error: require File.expand_path('../config/environment', __dir__)

FrozenError:
  can't modify frozen Array: [...]

# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/engine.rb:588:in `unshift'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/engine.rb:588:in `block in <class:Engine>'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:32:in `instance_exec'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:32:in `run'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:61:in `block in run_initializers'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:50:in `each'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:50:in `tsort_each_child'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/initializable.rb:60:in `run_initializers'
# /Users/mattiagiuffrida/.rvm/gems/ruby-3.1.2/gems/railties-6.1.6.1/lib/rails/application.rb:391:in `initialize!'
# ./config/environment.rb:7:in `<main>'
```

If I `require File.expand_path('../config/environment', __dir__)` before `stimpack` (i.e. add `--require rails_helper` to my `.rspec` BEFORE `stimpack/rspec`), then everything works fine.
Although this solution is relatively easy, we have so far preferred only to require `spec_helper` by default as this significantly speeds up individual tests execution for non-rails specs (e.g. `lib` classes).

**Comment by baweaver** [comment]
Could you run this with `--fail-fast` and share the output?

**Comment by iMacTia** [comment]
No change at all unfortunately.
I believe `--fail-fast` stops the run on the first failing test, but in my case these errors happen while loading the test suite 😕

**Comment by ngan** [comment]
Hey @iMacTia, in your `.rspec`, can you just have:

```
--require spec_helper
--require stimpack/rspec
```

I'm assuming your test files require `rails_helper` when they need it...in which case, Rails would boot. `stimpack/rspec` doesn't need Rails to do its job.

**Comment by iMacTia** [comment]
@ngan If I only require the `spec_helper` in `.rspec`, I get the error above.

> I'm assuming your test files require rails_helper when they need it

That's exactly what causes the issue. Requiring `rails_helper` loads the environment, which runs the initialisation and shows the `FrozenError`. If I require `rails_helper` in `.rspec` instead, the error doesn't show. Which makes me think it's somehow related to the ordering.

Also, requiring `spec_helper` but without `stimpack/rspec` in `.rspec` works just fine.

To recap:

```
# This works (we've been doing this until now)
--require spec_helper

# This doesn't work, causes the FrozenError
--require spec_helper
--require stimpack/rspec

# This works, it's what we're currently using, although some individual tests are now a bit slower because we need to load the Rails environment even when not necessary
--require rails_helper
--require stimpack/rspec
```

**Comment by alexevanczuk** [comment]
Hi @iMacTia – are you still experiencing this issue from a while back? I'm wondering if we can get to the bottom of this or close this issue out if it's no longer an issue!

I think these days all you need to do is add `--require packs/rails/rspec` to your `.rspec`. If that works just fine for you we can go ahead and close this out as well.

**Comment by alexevanczuk** [comment]
Thanks for confirming @iMacTia .

Please let me know if you or anyone else reading this has further problems and we can reopen this or file a new issue.
